tensorflow_datasets/image/mnist.py

```diff
@@ -124,8 +125,15 @@ def _split_generators(self, dl_manager):
         "test_data": _MNIST_TEST_DATA_FILENAME,
         "test_labels": _MNIST_TEST_LABELS_FILENAME,
     }
-    mnist_files = dl_manager.download_and_extract(
-        {k: urllib.parse.urljoin(self.URL, v) for k, v in filenames.items()})
+
+    try:
```

Comment:
the urls data have same folder structure so we can use try..exception block for just a `dl_manager.download()`, then use the `dl_manager.extract` method. 

And you need to be more clear when catch exception, Exception is too general for dl_manager. You should just catch the `tfds.core.download.downloader.DownloadError`.

Reply:
Thank you, fix it